### PR TITLE
TOC and Index skeleton for AI docs

### PR DESCRIPTION
**Pull request by haywoodsloan**
## Summary

Adds an initial TOC and index page for the AI docs. 

This is just a skeleton and will be expanded as we complete more docs. 

Additionally, we'll handle reorganization after the docs set is complete.

### CODE REVIEW

**Review by catbutler — APPROVED** [review]
Super! Thanks for creating this. Please LMK if I should add my docs/items.

**Review by catbutler — APPROVED** [review]

File: docs/index.yml [hunk 1]
```diff
@@ -321,6 +321,8 @@ additionalContent:
         # Card
         - title: Machine learning and AI
```
Comment (gewarren) [inline docs/index.yml:322]:
```suggestion
        - title: AI and machine learning
```

File: docs/index.yml [hunk 2]
```diff
@@ -321,6 +321,8 @@ additionalContent:
...
           links:
+            - url: ai/index.yml
+              text: AI for .NET Developers
```
Comment (gewarren) [inline docs/index.yml:325]:
```suggestion
              text: AI for .NET developers
```

**Review by gewarren — APPROVED** [review]
Looks good.

**Pull request merged**